### PR TITLE
[Kids Profile] Thank You screen

### LINKED ISSUE #1937: [Kids profile] Build first screen

### PR DESCRIPTION
**Pull request by danielebogo**
| 📘 Part of: #1935 
|:---:|

Fixes #1937 

This PR introduces the next screen prompted by tapping the Kids Profile banner.

## To test

> [!NOTE]
> You need to enable the kidsProfile Feature Flag

- CI must be 🟢 
- Run this branch and go to your profile page
- You should see the Kids Profile banner
- Tap on `Request Early Access`
- You should see a sheet being prompted
- If you tap on `No, Thank You` the sheet will be dismissed
- If you tap on `Send Feedback` the sheet should expand and a placeholder should appear

https://github.com/user-attachments/assets/4f3fe26e-4fc4-4162-b7fc-5f8afc9fdc1e

## Checklist

- [ ] I have considered if this change warrants user-facing release notes and have added them to `CHANGELOG.md` if necessary.
- [ ] I have considered adding unit tests for my changes.
- [ ] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: podcasts/Kids Profile/KidsProfileSheet.swift [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+struct KidsProfileSheet: View {
+    @ObservedObject var viewModel: KidsProfileSheetViewModel
+
+    @EnvironmentObject var theme: Theme
+
+    var body: some View {
+        VStack {
+            if viewModel.currentScreen == .submit {
+                submitScreen
+                    .transition(viewModel.transition)
```
Comment (leandroalonso) [inline podcasts/Kids Profile/KidsProfileSheet.swift]: I'm just curious: the `ViewModel` is, well, a view model, but why have the transition there? I feel like this is specific to this view.

No need to change, just curious about the reasoning behind it!

**Review by leandroalonso — APPROVED** [review]
✅ Tested with multiple themes
✅ Tested the mentioned steps

@danielebogo I tested with big font sizes and the card doesn't resize as a whole. This is not part of the scope of this task so not a blocker, just thought it worth mentioning it.

File: podcasts/Kids Profile/KidsProfileSheet.swift (hunk 1 above)
Comment (danielebogo) [reply podcasts/Kids Profile/KidsProfileSheet.swift]: @leandroalonso I thought to keep the view as much isolated as possible, but I moved the transition to the sheet container instead

**Pull request merged**